### PR TITLE
Fix test for gh5304 issue

### PR DESCRIPTION
**Pull request by mechanik20051988**
In previous test version on_replace trigger in
has never been called

### CODE REVIEW

**Comment by Korablev77** [comment]
LGTM but for git history could please provide more info: why this function was never called and why your changes fix the problem. Also you could use Follow-up tag in commit message:
Follow-up #5304

File: test/box/gh-5304-insert_during_recovery.test.lua [hunk 1]
```diff
@@ -18,6 +18,16 @@ test_run:cmd('start server test with args="replace" with crash_expected=True')
...
 test_run:cmd('start server test with args="upsert" with crash_expected=True')
 test_run:cmd('start server test with args="replace is_recovery_finished"')
+test_run:cmd('switch test')
+box.space.temp:select()
+box.space.loc:select()
+-- Creating a new space and index invoke on_replace trigger in _index space,
+-- which insert tuples in temp and loc spaces (see gh-5304-insert_during_recovery.lua)!
+s0 = box.schema.space.create('test')
+i0 = s0:create_index('test')
+box.space.temp:select()
```
Comment (Korablev77) [inline test/box/gh-5304-insert_during_recovery.test.lua:28]: Could you also add selects to other (insert/update) test cases to make sure that on_replace trigger was called?

**Comment by Korablev77** [comment]
I've also fixed commit message (just corrected several language mistakes):

In previous test version on_replace trigger (created on
_user sapace) is never called. It's because is_recovery_finished()
always returns false: on_schema_init is invoked BEFORE
user's data recovery process (so trigger is not created at all
at this moment).
In new test version you can see correct user case:
we create on_replace trigger on _index system space,
which replaces tuples in temp and loc spaces. So each time
user creates new space and index for it, trigger replaces
tuples in temp and loc spaces.

**Comment by Korablev77** [comment]
So now not only test itself is fixed, but also some implementation internals. Could you reflect this change in commit message as well? That is why we should compare memtx state with OK and not with FINAL_RECOVERY.

**Review by Korablev77 — APPROVED** [review]
LGTM

**Review by alyapunov — APPROVED** [review]

**Comment by Korablev77** [comment]
Merged into master, 2.7 and 2.6

**Pull request closed without merging**